### PR TITLE
Interactive CLI option for move_to_folder

### LINKED ISSUE #135: Proposal: move pack to folder
We’re currently moving many packs into folders. We were leveraging the `move_to_parent!` command in `bin/packs` and that helped us get going. However, `move_to_parent!` assumes you are moving a pack under a parent pack. This will then create a package.yml file if it doesn't exist. We don't want to do this.

I see two possible user experiences and three ways to implement this.

1) create a `move_to_folder` which does everything that `move_to_parent` does, but does not create a package.yml  

2) modify `move_to_parent` to take an argument about whether to create a package.yml

3) modify `move_to_parent` to never create a package.yml but modify it if it already exists.

## Existing User Experience

```
Existing options
‣ 1. Create a new pack
  2. Move files
  3. Add a dependency
  4. Get info on one or more packs
  5. Query violations about a pack
  6. Make files or directories public
  7. Move a child pack to be nested under a parent pack
  8. Rename a pack
  9. Run bin/packwerk check
  10. Run bin/packwerk update
  11. Run bin/packwerk validate (detects cycles)
  12. Lint packs/*/package.yml for one or more packs
```

## User Experience: Add a new option

Here's one possible user experience:

```
Existing options
‣ 1. Create a new pack
  2. Move files
  3. Move a pack <------------- new
  4. Add a dependency
  5. Get info on one or more packs
  6. Query violations about a pack
  7. Make files or directories public
  8. Move a child pack to be nested under a parent pack
  9. Rename a pack
  10. Run bin/packwerk check
  11. Run bin/packwerk update
  12. Run bin/packwerk validate (detects cycles)
  13. Lint packs/*/package.yml for one or more packs
```

```
Hello! What would you like to do? Move a pack
Please select a pack: packs/my_pack
Please input the destination directory: 
```

Pros
- I like how `move a pack` parallels with `move a file`

Cons
- I dislike that this will add another option to the bin/packs CLI command which already has 12 options.

## User Experience: repurpose an existing option

```
‣ 1. Create a new pack
  2. Move files
  3. Move a pack <------------- new
  4. Add a dependency
  5. Get info on one or more packs
  6. Query violations about a pack
  7. Make files or directories public
  8. Move a child pack to be nested under a parent pack <-------- deleted
  9. Rename a pack
  10. Run bin/packwerk check
  11. Run bin/packwerk update
  12. Run bin/packwerk validate (detects cycles)
  13. Lint packs/*/package.yml for one or more pack
```

```
Hello! What would you like to do? Move a pack to a folder or nested under a parent pack
Please select the pack to move: packs/my_pack
Would you like to 
‣ 1. Move to a folder
  2. Nest under a parent pack
 
# If picking 1
Please input the destination directory: 

# If picking 2
Please select the pack that will be the parent: 
```

I think *User Experience: repurpose an existing option*  is the winner.

Given this, we could rename `move_to_parent` to `move_pack` and have it never create a package.yml but modify it if it already exists.

### PR DESCRIPTION
**Pull request by schoblaska**
Adds an interactive CLI option for the `move_to_folder` command added in https://github.com/rubyatscale/packs/pull/136.

Both `move_to_folder` and `move_to_parent` are nested under a top-level CLI option "Move a pack."

If you attempt to move a pack into a folder that happens to be a pack itself, you get a warning message and are given the chance to use the `move_to_parent` command instead.

```
$ bin/packs
Hello! What would you like to do? (Press ↑/↓ arrow to move, Enter to select
nd letters to filter)
  1. Create a new pack
  2. Move files
‣ 3. Move a pack
  4. Add a dependency
  5. Get info on one or more packs
  6. Query violations about a pack
  7. Make files or directories public
  8. Rename a pack
  9. Run bin/packwerk check
  10. Run bin/packwerk update
  11. Run bin/packwerk validate (detects cycles)
  12. Lint packs/*/package.yml for one or more packs

Hello! What would you like to do? Move a pack
What do you want to do? (Press ↑/↓ arrow to move and Enter to select)
‣ Move a child pack to be nested under a parent pack
  Move a pack to a folder that is not a pack
  
$ bin/packs
Hello! What would you like to do? Move a pack
What do you want to do? Move a pack to a folder that is not a pack
Please select the pack that you want to move packs/foo
What directory do you want to move packs/foo to? packs/bar
The directory packs/bar contains a pack. Add foo as a dependency?
(Press ↑/↓ arrow to move and Enter to select)
‣ Yes
  No
 ```

Fixes #135

### CODE REVIEW

File: lib/packs/private/interactive_cli/pack_directory_selector.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# typed: strict
+
+module Packs
+  module Private
+    module InteractiveCli
+      class PackDirectorySelector
+        extend T::Sig
+
+        sig { params(prompt: TTY::Prompt, question_text: String).returns(String) }
+        def self.select(prompt, question_text: 'Select a directory')
```
Comment (perryqh) [inline lib/packs/private/interactive_cli/pack_directory_selector.rb:10]: Should we add some specs for this?

File: lib/packs/private/interactive_cli/pack_directory_selector.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+          directories = []
+
+          Packs::Specification.config.pack_paths.each do |path|
+            directories << Dir.glob(path).select { |f| File.directory? f }
+          end
```
Comment (perryqh) [inline lib/packs/private/interactive_cli/pack_directory_selector.rb:15]: Is there a way we can test this?

**Comment by perryqh** [comment]
Really solid PR! I just have a question about testing.

File: lib/packs/private/interactive_cli/pack_directory_selector.rb (hunk 2 above)
Comment (schoblaska) [reply lib/packs/private/interactive_cli/pack_directory_selector.rb:15]: Yes - good call. I added a test, and it caught that `directories` needed to be flattened 😅

**Review by ashleywillard — APPROVED** [review]

**Pull request merged**